server.go

```diff
@@ -131,8 +134,15 @@ func (svr *Server) sftpServerWorker(doneChan chan error) {
 			return
 		}
 
+		// If server is operating read-only and a write operation is requested,
+		// return permission denied
+		if dPkt.writing() && svr.readOnly {
+			_ = svr.sendPacket(statusFromError(dPkt.id(), syscall.EPERM))
```

Comment:
This is a habit from the way I explicitly ignore errors in my personal and work code.  It also stops `errcheck` from generating an error, since you are explicitly stating that you are aware an error is returned, but do not care about its result.

If you'd prefer to remove these, I can do that.